File: ajax-java/src/main/java/fr/isima/cours/jee/service/HourGenerator.java
package fr.isima.cours.jee.service;

import org.springframework.stereotype.Component;

import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

@Component
public class HourGenerator {

    static final String FORMAT_STANDARD_DATE = "HH:mm:ss 'le' dd/MM/yyyy";

    public Hour nowWithStandardFormat() {
        final LocalDateTime unformatted = LocalDateTime.now();
        return new Hour(unformatted, DateTimeFormatter.ofPattern(FORMAT_STANDARD_DATE).format(unformatted));
    }
}

File: ajax-kotlin/src/main/kotlin/fr/isima/service/HourGenerator.kt
package fr.isima.service

import org.springframework.stereotype.Component

import java.time.LocalDateTime
import java.time.format.DateTimeFormatter

const val FORMAT_STANDARD_DATE = "HH:mm:ss 'le' dd/MM/yyyy"

@Component
class HourGenerator {
    fun nowWithStandardFormat(): Hour =
        LocalDateTime.now().run {
            Hour(this, DateTimeFormatter.ofPattern(FORMAT_STANDARD_DATE).format(this))
        }
}

class Hour internal constructor(val unformatted: LocalDateTime, val formatted: String)
